### PR TITLE
SAK-48175  TRINITY: Sign In > Wrong Password > No error message

### PR DESCRIPTION
**Pull request by kunaljaykam**
https://sakaiproject.atlassian.net/browse/SAK-48175

### CODE REVIEW

File: login/login-render-engine-impl/impl/src/webapp/vm/defaultskin/xlogin.vm [hunk 1]
```diff
@@ -74,6 +75,23 @@
...
+                    for (const input of formFields) {
+                        input.classList.remove("is-invalid");
+                    }
+                });
+                input.addEventListener("click", () => {
+                    for (const input of formFields) {
+                        input.classList.remove("is-invalid");
+                    }
+                });
+            }
```
Comment (adrianfish) [inline login/login-render-engine-impl/impl/src/webapp/vm/defaultskin/xlogin.vm:93]:
```suggestion
            const formFields = document.querySelectorAll(".form-control");
            const removeInvalid = el => formFields.forEach(el => el.classList.remove("is-invalid"));

            formFields.forEach(input => {
            
                input.addEventListener("input", removeInvalid);
                input.addEventListener("click", removeInvalid);
            });
```

**Review by adrianfish — APPROVED** [review]
Approved, but I've added a suggested commit.

File: login/login-render-engine-impl/impl/src/webapp/vm/defaultskin/xlogin.vm [hunk 2]
```diff
@@ -74,6 +75,16 @@
...
             //Add an on-load listener to focus the password field
             window.addEventListener("load", () => document.querySelector("#eid")?.focus());
+
+            // Remove the the error class from the form fields when the user starts typing
+            const formFields = document.querySelectorAll(".form-control");
+            const removeInvalid = el => formFields.forEach(el => el.classList.remove("is-invalid"));
+            formFields.forEach(input => {
+            
+                input.addEventListener("input", removeInvalid);
+                input.addEventListener("click", removeInvalid);
```
Comment (stetsche) [inline login/login-render-engine-impl/impl/src/webapp/vm/defaultskin/xlogin.vm:85]: Maybe the once option on the eventlistener would make sense here, what do you think?
`input.addEventListener("click", removeInvalid, { once: true });`
I think this classes are not re-added?

**Review by stetsche — APPROVED** [review]
Looks good Kunal!

**Pull request merged**